src/sage/categories/enumerated_sets.py

```diff
@@ -457,10 +457,23 @@ def __getitem__(self, i):
                 [1]
                 sage: F[1::2]
                 [2]
+
+            TESTS:
+
+            Verify that an infinite index raises an error::
+
+                sage: F = FiniteEnumeratedSet([1,2,3,4,5])
+                sage: F[oo]
+                Traceback (most recent call last):
+                ...
+                TypeError: unable to coerce <class 'sage.rings.infinity.PlusInfinity'>
+                to an integer
             """
             from sage.rings.infinity import Infinity
+            from sage.rings.integer_ring import ZZ
             if isinstance(i, slice):
                 return self.unrank_range(i.start, i.stop, i.step)
+            i = ZZ(i)
```

Comment:
```suggestion
            if isinstance(i, slice):
                return self.unrank_range(i.start, i.stop, i.step)
            i = Integer(i)
```
Actually, `Integer` is already a top-level import, so let's use that.